### ISSUE TITLE
Make “Monte Carlo (Recommended)” the default method; keep “ML Optimizer (Legacy)” as option

### ISSUE THREAD

**Issue by rrlamichhane** [rrl-personal-projects/profit-taking-simulator#74]
**Goal**
Expose a new **Volatility-Based Monte Carlo** method as the **default** plan recommender and chart overlay. Keep the existing **LightGBM Optimizer** as a selectable **“ML Optimizer (Legacy)”** method. Both methods produce a **profit-taking plan** with four parameters:

* **Take-Profit Trigger**: % **above** prior ATH
* **Take-Profit Amount**: % of **total holdings**
* **Re-Invest Trigger**: % **below** ATH
* **Re-Invest Amount**: % of **last profit-taken**

This must remain **static (no backend)**, loading **precomputed artifacts** and responding instantly to **Horizon, CAGR, Annual Vol** inputs.

---

### Why this change

* Our primary use case is **probabilistic simulation** (how the asset *behaves*), not point prediction.
* The Monte Carlo engine **anchors to user CAGR**, reproduces **BTC-style volatility** (clustering + fat tails), and is **deterministic and fast** in the browser.
* ML remains useful as a reference, but it’s not the most aligned default for this task.

---

## UX / UI

**Method Dropdown (new)**

* **Default**: **Monte Carlo (Recommended)**
* Alternate: **ML Optimizer (Legacy)**

**Behavior**

* When **Monte Carlo (Recommended)** is selected:

  * The four plan fields are **auto-populated** from the MC recommender and shown as **recommended** (locked or read-only with a small “Recommended” badge).
  * Chart updates to show the **fan chart** (median + 68%/95% bands) from the MC engine and overlays the plan markers/events as today.
* When **ML Optimizer (Legacy)** is selected:

  * The four fields populate from the existing LightGBM logic (current behavior), shown with a **“Legacy (ML)”** badge.
  * Chart stays consistent; only the source of the plan changes.

**Existing controls** (unchanged)

* **Horizon** (years)
* **CAGR** (annual)
* **Annual Avg Volatility** (default 60%)

---

## Data & Artifacts (static)

We’ll ship two precomputed artifact families under `/public/precomp/`:

1. **Monte Carlo Fan Quantiles** (for the chart)

* Covers **CAGR grid**: **0–150%** in **5%** steps (31 points)
* **Hmax**: **520 weeks** (10 years) for weekly fan charts
* **Volatility handling**: either

  * **A)** Precompute across a small **vol grid** (e.g., 20–120% in 10% steps), **or**
  * **B)** Precompute at a calibrated base vol and **scale residuals** at runtime to match the selected annual vol (lighter payload; acceptable approximation)
* Quantiles stored in **log-price space**: `p05, p16, p50, p84, p95` per week × per CAGR (× per vol if using A)

2. **Monte Carlo Recommended Plan Table** (for thresholds/amounts)

* For each tuple `(horizonYears ∈ {1..10}, cagr ∈ grid above, annualVol ∈ volGrid or scaled)`, store the **recommended plan** that maximizes our target metric (see below) over MC paths:

  * `{ tpTriggerFromAthPct, tpAmountAsTotalPct, reinvestTriggerFromAthPct, reinvestAmountAsLastSalePct }`
  * Include `meta`: `{ objective, interpolated: bool, dataHash }`
* If you choose approach **B** (vol scaling at runtime), compute recommendations on a **vol grid** and **interpolate** between vol points at runtime.

**File examples**

* `/public/precomp/mc_fan.json` (or `.bin` + index)
* `/public/precomp/mc_recommendations.json`

**Metadata**

```json
{
  "version": 1,
  "dataHash": "sha256(...)",
  "weekConvention": "Sunday 00:00 UTC close",
  "cagrGrid": [0,5,10,...,150],
  "hmaxWeeks": 520,
  "volGrid": [20,30,...,120],  // if using vol grid (A)
  "objective": "maximize_expected_terminal_wealth", 
  "buildTime": "2025-10-17T..Z"
}
```

---

## Monte Carlo Engine (offline, precompute)

**Process (offline script; no code here, just behavior)**

* Use historical weekly BTC returns to calibrate **residual shape**:

  * **Drift**: set by user CAGR during simulation (so train on residuals after removing a nominal drift)
  * **Volatility**: **EWMA** (λ≈0.94 weekly) for clustering
  * **Innovations**: **Student-t** or **empirical bootstrap** of standardized residuals for fat tails
* For each CAGR on the grid (and each vol on volGrid if A):

  * Simulate many paths (e.g., 10k) to **520 weeks**
  * Store **weekly quantiles** (log-price) for the chart
  * Evaluate a modest **threshold grid** (e.g., TP trigger 5–80% in 5% steps, TP amount 5–50% in 5% steps, Re-invest trigger −5% to −60% in 5% steps, Re-invest amount 25–150% in 25% steps) on the generated paths:

    * **Objective**: `maximize_expected_terminal_wealth` (or `median_terminal_wealth`), with ties broken by fewer trades; record **expected trades** and **drawdown** as side metrics
  * Store the **argmax plan** for that (horizon, CAGR, vol) combo in the **Recommended Plan Table**

**Determinism**: Fixed seed per (cagr, vol, dataHash) so artifacts are reproducible.

---

## Frontend Integration (React + Recharts)

**State**

* Add `method: 'montecarlo' | 'ml_legacy'` (default `'montecarlo'`), persisted like other controls.
* Derived selector: `activePlan` returns the plan from the active method.

**Data loading**

* On app load, fetch and cache `mc_fan` and `mc_recommendations` artifacts.
* Keep existing ML artifacts/loading as is.

**Interpolation rules**

* **CAGR**: linear interpolation **in log-quantile space** between nearest grid points.
* **Vol**:

  * If **A (vol grid)**: linear interpolation between nearest vol grid points (quantiles & plan table).
  * If **B (vol scaling)**: scale residual variance at runtime for fan chart; for **plan table**, interpolate from a sparse vol grid baked into recommendations.
* **Horizon**: slice first `k weeks` (1–10y); for any longer horizons the app currently supports, keep existing behavior (but we do **not** expand weekly fan beyond 10y).

**Controls behavior**

* When **Monte Carlo (Recommended)** is active:

  * Populate 4 plan fields from `mc_recommendations` (interpolated if needed), mark as **Recommended**, and lock or visually denote read-only.
  * Recharts draws:

    * **p50** line and **68%/95%** bands from `mc_fan` (interpolated)
    * Any strategy markers/events already supported
* When **ML Optimizer (Legacy)** is active:

  * Populate fields from current LightGBM output; unlock editing if your current UX allows.

---

## Acceptance Criteria

**Default & UX**

* **Default method** on first load = **Monte Carlo (Recommended)**.
* Changing **Horizon, CAGR, Annual Vol** updates:

  * The fan chart (smoothly; no visible jumps across grid points).
  * The 4 plan fields (instantly; with a small “≈ interpolated” badge if not exact grid match).

**Determinism & Bounds**

* For a fixed `(horizon, cagr, vol, dataHash)`, the recommended plan and the fan chart are **identical** across reloads.
* Validation clamps:

  * `tpTriggerFromAthPct` ∈ [0%, 300%]
  * `tpAmountAsTotalPct` ∈ [0%, 100%]
  * `reinvestTriggerFromAthPct` ∈ [−95%, 0%]
  * `reinvestAmountAsLastSalePct` ∈ [0%, 200%]

**Performance**

* Artifacts load and parse in **<100 ms** on desktop; chart updates are snappy (no main-thread jank).
* No Web Workers required for this phase.

**Visual**

* Fan chart shows **p50** + shaded **68%/95%** bands; legend text: “Bands reflect simulated BTC-style volatility around your CAGR assumption.”
* Method badges: “Recommended” vs “Legacy (ML)”.

---

## Deliverables

* `/public/precomp/mc_fan.(json|bin)` and `/public/precomp/mc_recommendations.json` with metadata (`version`, `dataHash`, grids).
* `Method` dropdown + state plumbing (default `montecarlo`).
* Interpolation helpers (CAGR, Vol) with quantile-ordering enforcement after interpolation.
* Wiring for chart & controls to use **active plan**.
* Short docs: `README_montecarlo.md` describing artifact fields, interpolation, objective, rebuild script.

---

## Test Plan

**Unit**

* Interpolation returns ordered quantiles (`p05≤p16≤p50≤p84≤p95`) per week.
* Recommendations lookups are deterministic; `meta.interpolated` is set correctly.

**Integration**

* Switch Method: field values and chart update immediately; no stale state.
* Extremes: `(cagr=0% or 150%)`, `(vol=20% or 120%)`, `(horizon=1y or 10y)` produce valid outputs and render bands.

**Visual**

* Smooth slider movement: no perceptible jumps when crossing CAGR grid points.
* Badges display appropriately (“Recommended” for MC, “Legacy (ML)” for ML).

---

## Notes / Assumptions

* The existing ML Optimizer remains untouched; we only expose it under “Legacy (ML)”.
* For simplicity and payload size, prefer **vol scaling (B)** for the fan chart and **sparse vol grid** for plan recommendations (interpolated) unless you decide to precompute full vol grid (A).
* Week boundary is consistent with your current dataset (documented in artifact).
